pkg/rebuild/maven/infer.go

```diff
@@ -168,12 +168,63 @@ func getJarJDK(ctx context.Context, name, version string, mux rebuild.RegistryMu
 		return "", errors.Wrap(err, "reading manifest file")
 	}
 	for _, line := range strings.Split(string(manifestReader), "\n") {
-		if strings.HasPrefix(line, "Build-Jdk:") {
+		if strings.HasPrefix(line, "Build-Jdk:") || strings.HasPrefix(line, "Build-Jdk-Spec:") {
 			_, value, _ := strings.Cut(line, ":")
 			return strings.TrimSpace(value), nil
 		}
 	}
-	return "", nil
+	return approximateJarJDK(zipReader)
+}
+
+func approximateJarJDK(jarZip *zip.Reader) (string, error) {
```

Comment:
nit: this function name describes that we're guessing the jdk but we're doing that above, as well, so this serves as *another* approximation and so shouldn't be named by its purpose but its function. one option would be `inferJDKFromBytecode` as that communicates the content of the heuristic.